src/metabase/driver.clj

```diff
@@ -304,7 +304,7 @@
         java.math.BigInteger            :BigIntegerField
         java.sql.Date                   :DateField
         java.sql.Timestamp              :DateTimeField
-        java.util.Date                  :DateField
+        java.util.Date                  :DateTimeField
```

Comment:
we would be losing precision throughout the application because a `:DateField` is synonymous with only representing a day, month, and year, and hence has no associated time component.  So if a value has date+time such as `ISODate("2014-02-01T10:15:00Z")` but we set its base_type to `:DateField` we are effectively casting it to just "2014-02-01" and saying that we will be ignoring the time component.

is there a reason why you feel this is not an appropriate change?